llvm/utils/TableGen/SearchableTableEmitter.cpp

```diff
@@ -510,7 +538,8 @@ void SearchableTableEmitter::emitGenericTable(const GenericTable &Table,
 
   // Emit the declarations for the functions that will perform lookup.
   if (Table.PrimaryKey) {
-    emitLookupDeclaration(Table, *Table.PrimaryKey, OS);
+    auto &Index = Table.PrimaryKey;
```

Comment:
Why is this change needed?

Reply:
This was needed before when an additional argument `ShouldReturnRange` was being passed to emitLookupFunction. But since, it was not needed anymore, we can remove this extra assignment statement as well. I have pushed a new patchset fixing the same